log/logsink_test.go

```diff
@@ -358,17 +355,15 @@ func TestLogger_DeveloperLogsLevel(t *testing.T) {
 	const v = 2
 
 	buf := bytes.NewBuffer(nil)
-	logger := log.NewLogger("", ioutil.Discard, v, log.JSONEncoder{})
-	logger.SetOutput(buf)
+	ls := log.NewLogSink("", ioutil.Discard, v, log.JSONEncoder{})
+	ls.SetOutput(buf)
 
 	msg := "hello, world"
+	logger := logr.New(ls)
+
 	logger.Info(msg)
 
-	if buf.Len() == 0 {
-		t.Fatal("expected log output, but buffer was empty")
-	}
-	assert.Contains(t, string(buf.Bytes()), fmt.Sprintf(`%q:%q`, log.MessageKey, msg))
-	assert.Contains(t, string(buf.Bytes()), fmt.Sprintf(`%q`, log.FileLineKey))
+	assert.Empty(t, buf, "expected NO logs to be present")
```

Comment:
Same here the previous implementation looks fine for me. Your verbosity level is 2 and the level of the test message is 1, means the message should be included in the buffer because (verbosity >= loglevel).